### ISSUE TITLE
[Bug]: Milvus cannot delete old data from Rocksdb, resulting in increasing disk storage

### ISSUE THREAD

**Issue by 1041787154** [milvus-io/milvus#23011]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.3
- Deployment mode(standalone or cluster):standalone 
- MQ type(rocksmq, pulsar or kafka):    rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus  2.2.2
- OS(Ubuntu or CentOS): Ubuntu 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Milvus cannot delete old data from Rocksdb, resulting in increasing disk storage

### Expected Behavior

Ubuntu

**Comment by yanliang567** [comment]
@1041787154 how  did you delete old data by milvus? In my opinions, Milvus does not delete data from rocksdb directly. please share a reproduce step or scripts for us.

/assign @1041787154 
/unassign

**Comment by xiaofan-luan** [comment]
try to change rocksmq.rocksmqPageSize to 256MB or 64MB see if it helps.
Thanks

**Comment by xiaofan-luan** [comment]
you have to reinsert your data to make this work

**Comment by xiaofan-luan** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Environment
> ```gfm
> - Milvus version:2.2.3
> - Deployment mode(standalone or cluster):standalone 
> - MQ type(rocksmq, pulsar or kafka):    rocksmq
> - SDK version(e.g. pymilvus v2.0.0rc2):pymilvus  2.2.2
> - OS(Ubuntu or CentOS): Ubuntu 
> - CPU/Memory: 
> - GPU: 
> - Others:
> ```
> 
> ### Current Behavior
> Milvus cannot delete old data from Rocksdb, resulting in increasing disk storage
> 
> ### Expected Behavior
> Ubuntu
> 
> ### Steps To Reproduce
> _No response_
> 
> ### Milvus Log
> _No response_
> 
> ### Anything else?
> _No response_

1. how large is your rdb data dir? 
2. how many collections did you create?

Generally, rdb takes 256MB * 2 * collection number * shard number disk space.
Thus if your has many collections, rocksdb may use 10s of GB disk spaces.

If that's far more than that, pls offer logs so we can take a look into it.

**Comment by xiaofan-luan** [comment]
> @1041787154 how did you delete old data by milvus? In my opinions, Milvus does not delete data from rocksdb directly. please share a reproduce step or scripts for us.
> 
> /assign @1041787154 /unassign

Guess the [1041787154](https://github.com/1041787154) wants to say garbage collection is not working

**Comment by xiaofan-luan** [comment]
@aoiasd  is this fixed in 2.2.9?
